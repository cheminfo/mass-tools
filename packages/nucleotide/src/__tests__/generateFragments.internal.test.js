import { generateFragments } from '..';

describe('test generate internal fragments', () => {
  it('nucleotide HODampDtmpDcmpDgmpDampH internal abw fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      abw: true,
    });
    expect(fragments).toStrictEqual([
      'HODtmpFurp$w4:a3-B',
      'HODtmpDcmpFurp$w4:a4-B',
      'HODcmpFurp$w3:a4-B',
    ]);
  });

  it('nucleotide HODampDtmpDcmpH internal aby fragments', () => {
    let fragments = generateFragments('HODthyDcmpDcmpDgmpDtmpDcmpH', {
      aby: true,
    });
    expect(fragments).toStrictEqual([
      'O-2P-1DcmpFurp$y5:a3-B',
      'O-2P-1DcmpDcmpFurp$y5:a4-B',
      'O-2P-1DcmpDcmpDgmpFurp$y5:a5-B',
      'O-2P-1DcmpFurp$y4:a4-B',
      'O-2P-1DcmpDgmpFurp$y4:a5-B',
      'O-2P-1DgmpFurp$y3:a5-B',
    ]);
  });

  it('nucleotide HODthyDampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpDampDcmpH internal abw fragments', () => {
    let fragments = generateFragments(
      'HODthyDampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpDampDcmpH',
      {
        abw: true,
      },
    );
    expect(fragments).toMatchSnapshot();
  });

  it('nucleotide HODampDtmpDcmpH internal aw fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      aw: true,
    });
    expect(fragments).toStrictEqual([
      'HODtmpDcmpO-1H-1$w4:a3',
      'HODtmpDcmpDgmpO-1H-1$w4:a4',
      'HODcmpDgmpO-1H-1$w3:a4',
    ]);
  });

  it('nucleotide HODampDtmpDcmpH internal bw fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      bw: true,
    });
    expect(fragments).toStrictEqual([
      'HODtmpDcmpH$w4:b3',
      'HODtmpDcmpDgmpH$w4:b4',
      'HODcmpDgmpH$w3:b4',
    ]);
  });

  it('nucleotide to sequence internal modification of HODamp(NH2)DtmpDcmpH with internal fragments', () => {
    let fragments = generateFragments('HODampDampDtmp(NH2)DampH', {
      abw: true,
    });
    expect(fragments).toStrictEqual(['HODampFurp$w3:a3-B']);
  });
});
