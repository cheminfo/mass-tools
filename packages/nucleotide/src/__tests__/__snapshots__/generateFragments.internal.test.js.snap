// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generate internal fragments nucleotide HODthyDampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpDampDcmpH internal abw fragments 1`] = `
Array [
  "HODampFurp$w12:a3-B",
  "HODampDcmpFurp$w12:a4-B",
  "HODampDcmpDgmpFurp$w12:a5-B",
  "HODampDcmpDgmpDtmpFurp$w12:a6-B",
  "HODampDcmpDgmpDtmpDgmpFurp$w12:a7-B",
  "HODampDcmpDgmpDtmpDgmpDcmpFurp$w12:a8-B",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpFurp$w12:a9-B",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpDampFurp$w12:a10-B",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpDampDampFurp$w12:a11-B",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpFurp$w12:a12-B",
  "HODcmpFurp$w11:a4-B",
  "HODcmpDgmpFurp$w11:a5-B",
  "HODcmpDgmpDtmpFurp$w11:a6-B",
  "HODcmpDgmpDtmpDgmpFurp$w11:a7-B",
  "HODcmpDgmpDtmpDgmpDcmpFurp$w11:a8-B",
  "HODcmpDgmpDtmpDgmpDcmpDcmpFurp$w11:a9-B",
  "HODcmpDgmpDtmpDgmpDcmpDcmpDampFurp$w11:a10-B",
  "HODcmpDgmpDtmpDgmpDcmpDcmpDampDampFurp$w11:a11-B",
  "HODcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpFurp$w11:a12-B",
  "HODgmpFurp$w10:a5-B",
  "HODgmpDtmpFurp$w10:a6-B",
  "HODgmpDtmpDgmpFurp$w10:a7-B",
  "HODgmpDtmpDgmpDcmpFurp$w10:a8-B",
  "HODgmpDtmpDgmpDcmpDcmpFurp$w10:a9-B",
  "HODgmpDtmpDgmpDcmpDcmpDampFurp$w10:a10-B",
  "HODgmpDtmpDgmpDcmpDcmpDampDampFurp$w10:a11-B",
  "HODgmpDtmpDgmpDcmpDcmpDampDampDtmpFurp$w10:a12-B",
  "HODtmpFurp$w9:a6-B",
  "HODtmpDgmpFurp$w9:a7-B",
  "HODtmpDgmpDcmpFurp$w9:a8-B",
  "HODtmpDgmpDcmpDcmpFurp$w9:a9-B",
  "HODtmpDgmpDcmpDcmpDampFurp$w9:a10-B",
  "HODtmpDgmpDcmpDcmpDampDampFurp$w9:a11-B",
  "HODtmpDgmpDcmpDcmpDampDampDtmpFurp$w9:a12-B",
  "HODgmpFurp$w8:a7-B",
  "HODgmpDcmpFurp$w8:a8-B",
  "HODgmpDcmpDcmpFurp$w8:a9-B",
  "HODgmpDcmpDcmpDampFurp$w8:a10-B",
  "HODgmpDcmpDcmpDampDampFurp$w8:a11-B",
  "HODgmpDcmpDcmpDampDampDtmpFurp$w8:a12-B",
  "HODcmpFurp$w7:a8-B",
  "HODcmpDcmpFurp$w7:a9-B",
  "HODcmpDcmpDampFurp$w7:a10-B",
  "HODcmpDcmpDampDampFurp$w7:a11-B",
  "HODcmpDcmpDampDampDtmpFurp$w7:a12-B",
  "HODcmpFurp$w6:a9-B",
  "HODcmpDampFurp$w6:a10-B",
  "HODcmpDampDampFurp$w6:a11-B",
  "HODcmpDampDampDtmpFurp$w6:a12-B",
  "HODampFurp$w5:a10-B",
  "HODampDampFurp$w5:a11-B",
  "HODampDampDtmpFurp$w5:a12-B",
  "HODampFurp$w4:a11-B",
  "HODampDtmpFurp$w4:a12-B",
  "HODtmpFurp$w3:a12-B",
]
`;
